spring-cloud-stream-binder-kinesis-core/src/main/java/org/springframework/cloud/stream/binder/kinesis/provisioning/KinesisTopicProvisioner.java

```diff
@@ -0,0 +1,135 @@
+package org.springframework.cloud.stream.binder.kinesis.provisioning;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.cloud.stream.binder.ExtendedConsumerProperties;
+import org.springframework.cloud.stream.binder.ExtendedProducerProperties;
+import org.springframework.cloud.stream.binder.kinesis.properties.KinesisBinderConfigurationProperties;
+import org.springframework.cloud.stream.binder.kinesis.properties.KinesisConsumerProperties;
+import org.springframework.cloud.stream.binder.kinesis.properties.KinesisProducerProperties;
+import org.springframework.cloud.stream.provisioning.ConsumerDestination;
+import org.springframework.cloud.stream.provisioning.ProducerDestination;
+import org.springframework.cloud.stream.provisioning.ProvisioningException;
+import org.springframework.cloud.stream.provisioning.ProvisioningProvider;
+
+/**
+ * 
+ * @author Peter Oates
+ *
+ */
+public class KinesisTopicProvisioner implements ProvisioningProvider<ExtendedConsumerProperties<KinesisConsumerProperties>,
+ExtendedProducerProperties<KinesisProducerProperties>>, InitializingBean {
+
+	private final static Logger log = LoggerFactory.getLogger(KinesisTopicProvisioner.class);
+	
+	private final KinesisBinderConfigurationProperties configurationProperties;
+	
+	public KinesisTopicProvisioner(KinesisBinderConfigurationProperties kinesisBinderconfigurationProperties) {
+		this.configurationProperties = kinesisBinderconfigurationProperties;	
+	}
+
+	@Override
+	public void afterPropertiesSet() throws Exception {
+		// TODO Auto-generated method stub
+		
+	}
+
+	@Override
+	public ProducerDestination provisionProducerDestination(String name,
+			ExtendedProducerProperties<KinesisProducerProperties> properties) throws ProvisioningException {
+		
+		KinesisProducerDestination producer = new KinesisProducerDestination(name);
+		log.info(String.format("Producer: %s", producer));
```

Comment:
Right. "Remove them" is good way to go, too.
But no unconditional less then ERROR logs, that's true